### PR TITLE
chore: Bump to kubernetes 1.31.0

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
Uses kubernetes 1.31 client libs

**This change breaks a stable metric** - I believe this is justified as the metric label never gave a correct result.
https://kubernetes.io/blog/2024/07/19/kubernetes-1-31-upcoming-changes/#deprecation-of-status-nodeinfo-kubeproxyversion-field-for-nodes-kep-4004-https-github-com-kubernetes-enhancements-issues-4004

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
/lgtm 
/hold
feel free to unhold

**Comment by mrueg** [comment]
@CatherineF-dev can you take another look? Thanks!

**Comment by dgrisonnet** [comment]
I think it might be better to follow the same deprecation plan as upstream. Removing it directly might be harmless here, but at least following the same plan as Kubernetes would set a precedent for what we should do in the future when such thing happen. We could make a deprecation notice in the next release changelog and then fully remove the label once the feature is removed?

File: docs/metrics/cluster/node-metrics.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
-| kube_node_info               | Gauge       | Information about a cluster node                                                                                          |                                                                                                                                                                                          | `node`=&lt;node-address&gt; <br> `kernel_version`=&lt;kernel-version&gt; <br> `os_image`=&lt;os-image-name&gt; <br> `container_runtime_version`=&lt;container-runtime-and-version-combination&gt; <br> `kubelet_version`=&lt;kubelet-version&gt; <br> `kubeproxy_version`=&lt;kubeproxy-version&gt; <br> `pod_cidr`=&lt;pod-cidr&gt; <br> `provider_id`=&lt;provider-id&gt; <br> `system_uuid`=&lt;system-uuid&gt; <br> `internal_ip`=&lt;internal-ip&gt; | STABLE       |
```
Comment (logicalhan) [inline docs/metrics/cluster/node-metrics.md:6]: I would rather preserve API stability but just enter a constant "disabled" value for the existing label.

**Comment by mrueg** [comment]
I made the requested changes, please have another look.

**Comment by mrueg** [comment]
/hold cancel 

@CatherineF-dev @rexagod @dgrisonnet friendly ping on this one :)

**Comment by rexagod** [comment]
/lgtm
/hold

Holding for others, feel free to unhold.

**Comment by mauri870** [comment]
Just a friendly reminder about the missing reviews so we can get this merged. Thanks!

**Comment by mrueg** [comment]
/hold cancel

I'd say this has been waiting long enough. If people have additional concerns, they can follow up with their own PR.

**Pull request merged**